File: packages/kotlin/src/main/kotlin/com/submitcms/sdk/SubmitCms.kt
package com.submitcms.sdk

import com.submitcms.sdk.modules.Addresses
import com.submitcms.sdk.modules.Ai
import com.submitcms.sdk.modules.Auth
import com.submitcms.sdk.modules.Billing
import com.submitcms.sdk.modules.Cart
import com.submitcms.sdk.modules.Categories
import com.submitcms.sdk.modules.ContentTypes
import com.submitcms.sdk.modules.CustomerOrders
import com.submitcms.sdk.modules.Delivery
import com.submitcms.sdk.modules.Locales
import com.submitcms.sdk.modules.Menus
import com.submitcms.sdk.modules.Orders
import com.submitcms.sdk.modules.Partner
import com.submitcms.sdk.modules.Payments
import com.submitcms.sdk.modules.Platform
import com.submitcms.sdk.modules.Records
import com.submitcms.sdk.modules.Schema
import com.submitcms.sdk.modules.Shopping
import com.submitcms.sdk.modules.Storage
import com.submitcms.sdk.modules.System
import com.submitcms.sdk.modules.Tracking

/**
 * SubmitCMS SDK giriş noktası.
 *
 * ```kotlin
 * val sdk = SubmitCms(SubmitConfig(
 *     token = BuildConfig.SUBMIT_TOKEN,
 *     mode = SubmitMode.PRODUCTION,
 *     locale = "tr",
 * ))
 *
 * // Ziyaretçiye içerik — oturum gerekmez
 * val posts = sdk.delivery.records("blog", mapOf("per_page" to 10))
 *
 * // Panel işlemleri — önce giriş
 * sdk.auth.console("admin@site.com", "parola")
 * sdk.records.create("blog", mapOf(
 *     "data" to mapOf("baslik" to "Merhaba"),
 *     "status" to "published",
 * ))
 * ```
 *
 * Tüm çağrılar `suspend`'dir ve IO dispatcher'a geçer — ana iş parçacığından
 * güvenle çağrılabilir.
 *
 * | Ne yapıyorsunuz | Modül | Kimlik |
 * |---|---|---|
 * | Uygulamada içerik gösterme | `delivery` | site token'ı |
 * | İçerik yazma | `records`, `contentTypes`, `menus` | token + oturum |
 * | Mağaza sepeti | `cart` | site token'ı |
 * | Sipariş yönetimi | `orders` | token + oturum + `orders` modülü |
 * | Müşteri kendi sitesini yönetiyor | `platform` | token + oturum + üyelik |
 * | Bayi paneli | `partner` | partner oturumu |
 */
class SubmitCms(val client: SubmitClient) {

    constructor(config: SubmitConfig) : this(SubmitClient(config))

    val auth = Auth(client)
    val delivery = Delivery(client)

    val records = Records(client)
    val contentTypes = ContentTypes(client)
    val categories = Categories(client)
    val locales = Locales(client)
    val schema = Schema(client)
    val menus = Menus(client)

    val cart = Cart(client)
    val shopping = Shopping(client)
    val orders = Orders(client)
    val myOrders = CustomerOrders(client)
    val addresses = Addresses(client)
    val payments = Payments(client)
    val billing = Billing(client)

    val platform = Platform(client)
    val partner = Partner(client)

    val ai = Ai(client)
    val storage = Storage(client)
    val tracking = Tracking(client)
    val system = System(client)

    /** Oturum token'ını elle yazar — saklanan JWT'yi geri yüklerken. */
    fun setAuthToken(token: String?) = client.setAuthToken(token)

    /** Aktif siteyi değiştirir (`EnvToken`). Çok siteli uygulamalarda. */
    fun setEnvironment(token: String?) = client.setEnvironment(token)

    fun setLocale(locale: String) = client.setLocale(locale)

    /** Misafir sepeti kimliği. */
    fun setGuestId(guestId: String?) = client.setGuestId(guestId)
}

File: packages/swift/Sources/SubmitCMS/SubmitCMS.swift
import Foundation

/// SubmitCMS SDK giriş noktası.
///
/// ```swift
/// import SubmitCMS
///
/// let sdk = SubmitCMS(config: .init(
///     mode: .production,
///     token: ProcessInfo.processInfo.environment["SUBMIT_TOKEN"]!,
///     locale: "tr"
/// ))
///
/// // Ziyaretçiye içerik — oturum gerekmez
/// let posts = try await sdk.delivery.records("blog", params: ["per_page": 10])
///
/// // Panel işlemleri — önce giriş
/// try await sdk.auth.console(email: "admin@site.com", password: "parola")
/// try await sdk.records.create("blog", payload: [
///     "data": ["baslik": "Merhaba"],
///     "status": "published",
/// ])
/// ```
///
/// Hangi modülü ne zaman:
///
/// | Ne yapıyorsunuz | Modül | Kimlik |
/// |---|---|---|
/// | Uygulamada içerik gösterme | `delivery` | site token'ı |
/// | İçerik yazma | `records`, `contentTypes`, `menus` | token + oturum |
/// | Mağaza sepeti | `cart` | site token'ı |
/// | Sipariş yönetimi | `orders` | token + oturum + `orders` modülü |
/// | Müşteri kendi sitesini yönetiyor | `platform` | token + oturum + üyelik |
/// | Bayi paneli | `partner` | partner oturumu |
public struct SubmitCMS: Sendable {
    /// Alt seviye HTTP istemcisi — başlık ve oturum denetimi için.
    public let client: SubmitClient

    public let auth: AuthModule
    public let delivery: DeliveryModule

    public let records: RecordsModule
    public let contentTypes: ContentTypesModule
    public let categories: CategoriesModule
    public let locales: LocalesModule
    public let schema: SchemaModule
    public let menus: MenusModule
    /// Destek talepleri gelen kutusu (panel tarafı; gönderim `delivery`de).
    public let tickets: TicketsModule

    public let cart: CartModule
    public let shopping: ShoppingModule
    public let orders: OrdersModule
    public let myOrders: CustomerOrdersModule
    public let addresses: AddressesModule
    public let payments: PaymentsModule
    public let billing: BillingModule

    public let platform: PlatformModule
    public let partner: PartnerModule

    public let ai: AiModule
    public let storage: StorageModule
    public let tracking: TrackingModule
    public let system: SystemModule

    public init(config: SubmitConfig, session: URLSession = .shared) {
        self.init(client: SubmitClient(config: config, session: session))
    }

    public init(client: SubmitClient) {
        self.client = client

        auth = AuthModule(client: client)
        delivery = DeliveryModule(client: client)

        records = RecordsModule(client: client)
        contentTypes = ContentTypesModule(client: client)
        categories = CategoriesModule(client: client)
        locales = LocalesModule(client: client)
        schema = SchemaModule(client: client)
        menus = MenusModule(client: client)
        tickets = TicketsModule(client: client)

        cart = CartModule(client: client)
        shopping = ShoppingModule(client: client)
        orders = OrdersModule(client: client)
        myOrders = CustomerOrdersModule(client: client)
        addresses = AddressesModule(client: client)
        payments = PaymentsModule(client: client)
        billing = BillingModule(client: client)

        platform = PlatformModule(client: client)
        partner = PartnerModule(client: client)

        ai = AiModule(client: client)
        storage = StorageModule(client: client)
        tracking = TrackingModule(client: client)
        system = SystemModule(client: client)
    }

    /// Oturum token'ını elle yazar — saklanan JWT'yi geri yüklerken.
    public func setAuthToken(_ token: String?) async {
        await client.setAuthToken(token)
    }

    /// Aktif siteyi değiştirir (`EnvToken`). Çok siteli uygulamalarda.
    public func setEnvironment(_ token: String?) async {
        await client.setEnvironment(token)
    }

    public func setLocale(_ locale: String) async {
        await client.setLocale(locale)
    }

    /// Misafir sepeti kimliği.
    public func setGuestId(_ guestId: String?) async {
        await client.setGuestId(guestId)
    }
}
